### ISSUE TITLE
Rationalize notifications (was: Notification inconsistencies)

### ISSUE THREAD

**Issue by mjtko** [alces-software/alces-flight-center#507]
We should consider notifications in general (will log an issue elsewhere regarding this), but at the moment there are a few inconsistencies which I consider bugs.

Current behaviours
-----

Action | Email? | Slack? 
---------|-----------|---------
New case | Yes to all except user creating case | Yes | 
Change case assignee | Yes to assignee only | Yes | 
Change case subject | Yes | No | 
Change case issue | Yes | No | 
Comment on case | Yes to all except user making comment | Yes |
Change case associations | No | Yes |
Case escalation | No | No |
Maintenance window state transition | Yes | Yes |
Maintenance ending soon | Yes, to all admins only | Yes |
CR creation | No | No |
CR state change (propose, decline, authorise, handover, complete, request changes) | Yes | Yes |
CR cancel | No | No |
Log created | No | Yes |

**Comment by jamesremuscat** [comment]
Is your expected behaviour that all of the things listed generate both types of notification? (Just for clarity...)

**Comment by mjtko** [comment]
Yup, I'd expect Slack notifications if there are emails. And I'd expect an email for CR cancellation.

**Comment by jamesremuscat** [comment]
We don't generate emails or Slack notifications for change request creation, either.

As part of work planned for 2018.7 we'll be doing an audit of all places we currently send emails and Slack notifications, so I expect we'll discover more soon.

**Comment by mjtko** [comment]
I looked for somewhere to log things to look at during the notification audit, but I couldn't find one, so here seems as good a place as any to start adding them. 😄 

 * Send a Slack notification on completion of daily checks for a cluster

**Comment by mjtko** [comment]
Thanks @jamesremuscat -- I'll go through the list and work out who/when/how we want to notify for the above conditions and let you know in a follow up comment.

**Comment by mjtko** [comment]
Ok, here's my table of notifications I think we should send:

| Event                                 | Email: site contacts | Email: support team | Slack |
|---------------------------------------|----------------------|---------------------|-------|
| New case (by site contact)            | ✓                    | ✓                   | ✓     |
| New case (by admin)                   | ✓                    | ✓                   | ✓     |
| Change case engineer ("assignee")     |                      | ✓                   | ✓     |
| Change case subject                   | ✓                    | ✓                   | ✓     |
| Change case issue                     |                      | ✓                   | ✓     |
| Change case associations              |                      | ✓                   | ✓     |
| Add case comment                      | ✓                    | ✓                   | ✓     |
| Escalate support tier                 |                      |                     |       |
| Set case time worked                  |                      |                     |       |
| Case resolution                       |                      | ✓                   | ✓     |
| Case closure                          |                      |                     |       |
| Maintenance window state transition   | ✓                    | ✓                   | ✓     |
| Maintenance ending soon               |                      | ✓                   | ✓     |
| Change request: draft creation        |                      | ✓                   | ✓     |
| Change request: customer state change | ✓                    | ✓                   | ✓     |
| Change request: draft cancellation    |                      | ✓                   | ✓     |
| Log entry creation                    |                      | ✓                   | ✓     |
| Daily check completion                |                      | ✓                   | ✓     |

Comments:

1. I think we should drop the idea of emailing admins to their own addresses (if we do currently, not sure), and solely use the general `tickets@` group for "support team" emails.

2. I don't think there's any harm in sending out an email to the site contact when they create a new case or when they add a comment.  In days of yore, when using an email support system, this was probably less desirable than it is when using the modern web-based one.  So, I think we can drop the removal of the creator/commenter address in those cases too.

3. As I've been looking through this, I've arrived at the conclusion that "support team" notifications should go both via email and via Slack.  I don't think there's any point in having notifications that are Slack- or email-only.

4. As we're currently using "assignee" for engineers only (and I have another issue to log regarding this), and as we're currently used to seeing the change of assignee in both Slack and in email, I think we can drop the "assignee only" requirement on (re)assignment emails.

5. There's a new notification for case resolution that I think would be useful to report to the support team -- my future self also thinks that there will be other categories here when/if we implement auto-resolution/yes/no/thanks workflow things (#397, #208).

6. There's a gap in the middle for escalation/time-worked/closure -- I don't think we need notifications for those, but I'm adding them here for completeness and in case anybody thinks we should (now or in future!).

**Comment by atoghill** [comment]
> 1. I think we should drop the idea of emailing admins to their own addresses (if we do currently, not sure), and solely use the general tickets@ group for "support team" emails.

@mjtko In regards to this comment whenever an email is sent out the tickets@ group should already be getting a copy. Are you wanting to solely use this for ST emails, doing away with a bcc address?

**Comment by mjtko** [comment]
@atoghill: keep the BCC address for every email that goes out. Think what I'm trying to say in (1) is that there's no need for any special rules that target admins in particular, cos they'll already get the email via `tickets@` as a BCC recipient.

**Comment by atoghill** [comment]
@mjtko Okay gotcha, I did wonder if that was the case but for the notifications that only go to the support team it sounded strange to me to only send an email to the bcc address

**Comment by mjtko** [comment]
Well, whatever makes sense - we currently only have CC recipients (and the BCC recipient) which also doesn't make much sense to me - ref #456 :smile: 

I guess if there are no addressees other than `tickets@`, then it could become a To recipient.  Changes only required if things don't work and/or don't take long to implement tho.

**Comment by jamesremuscat** [comment]
I've just noticed that some aspects of this issue weren't covered by #546 so I'm reopening it.

In particular: 

2. I don't think there's any harm in sending out an email to the site contact when they create a new case or when they add a comment. In days of yore, when using an email support system, this was probably less desirable than it is when using the modern web-based one. So, I think we can drop the removal of the creator/commenter address in those cases too.

and

4. As we're currently using "assignee" for engineers only (and I have another issue to log regarding this), and as we're currently used to seeing the change of assignee in both Slack and in email, I think we can drop the "assignee only" requirement on (re)assignment emails.
